src/SignalR/server/SignalR/test/EndToEndTests.cs

```diff
@@ -532,6 +532,84 @@ private async Task ServerClosesConnectionWithErrorIfHubCannotBeCreated(HttpTrans
             }
         }
 
+        [Fact]
+        [LogLevel(LogLevel.Trace)]
+        public async Task UnauthorizedHubConnectionDoesNotConnectWithEndpoints()
+        {
+            bool ExpectedErrors(WriteContext writeContext)
+            {
+                return writeContext.LoggerName == typeof(HttpConnection).FullName &&
+                       writeContext.EventId.Name == "ErrorWithNegotiation";
+            }
+
+            using (StartServer<Startup>(out var server, ExpectedErrors))
+            {
+                var logger = LoggerFactory.CreateLogger<EndToEndTests>();
+
+                var url = server.Url + "/authHubEndpoints";
+                var connection = new HubConnectionBuilder()
+                        .WithLoggerFactory(LoggerFactory)
+                        .WithUrl(url, HttpTransportType.LongPolling)
+                        .Build();
+
+                try
+                {
+                    logger.LogInformation("Starting connection to {url}", url);
+                    await connection.StartAsync().OrTimeout();
+                    Assert.True(false);
+                }
+                catch (Exception ex)
+                {
+                    Assert.Equal("Response status code does not indicate success: 401 (Unauthorized).", ex.Message);
+                }
+                finally
+                {
+                    logger.LogInformation("Disposing Connection");
+                    await connection.DisposeAsync().OrTimeout();
+                    logger.LogInformation("Disposed Connection");
+                }
+            }
+        }
+
+        [Fact]
+        [LogLevel(LogLevel.Trace)]
+        public async Task UnauthorizedHubConnectionDoesNotConnect()
```

Comment:
Should this have a test that calling it with the requisite auth works?

Reply:
Yeah probably, I was going to and then forgot.